homeassistant/components/system_bridge/sensor.py

```diff
@@ -279,23 +368,27 @@ async def async_setup_entry(
             SystemBridgeSensor(coordinator, description, entry.data[CONF_PORT])
         )
 
-    for partition in coordinator.data.disk.partitions:
-        entities.append(
-            SystemBridgeSensor(
-                coordinator,
-                SystemBridgeSensorEntityDescription(
-                    key=f"filesystem_{partition.replace(':', '')}",
-                    name=f"{partition} space used",
-                    state_class=SensorStateClass.MEASUREMENT,
-                    native_unit_of_measurement=PERCENTAGE,
-                    icon="mdi:harddisk",
-                    value=lambda data, p=partition: getattr(
-                        data.disk, f"usage_{p}_percent", None
+    for index_device, device in enumerate(coordinator.data.disks.devices):
+        if device.partitions is None:
+            continue
+
+        for index_partition, partition in enumerate(device.partitions):
+            entities.append(
+                SystemBridgeSensor(
+                    coordinator,
+                    SystemBridgeSensorEntityDescription(
+                        key=f"filesystem_{partition.mount_point.replace(':', '')}",
+                        name=f"{partition.mount_point} space used",
+                        state_class=SensorStateClass.MEASUREMENT,
+                        native_unit_of_measurement=PERCENTAGE,
+                        icon="mdi:harddisk",
+                        value=lambda data,
```

Comment:
Please use a parenthesis around the lambda and indent it so it's easier to separate the lambda from the entity description parameters.